File: src/main/java/com/algorand/algosdk/v2/client/indexer/SearchForTransactions.java
package com.algorand.algosdk.v2.client.indexer;

import com.algorand.algosdk.v2.client.common.Client;
import com.algorand.algosdk.v2.client.common.Query;
import com.algorand.algosdk.v2.client.common.QueryData;
import com.algorand.algosdk.v2.client.common.Response;
import com.algorand.algosdk.v2.client.model.TransactionsResponse;


/**
 * Search for transactions. /transactions 
 */
public class SearchForTransactions extends Query {

	private String address;
	public String address() {
		return this.address;
	}
	private String addressRole;
	public String addressRole() {
		return this.addressRole;
	}
	private String afterTime;
	public String afterTime() {
		return this.afterTime;
	}
	private Long assetId;
	public Long assetId() {
		return this.assetId;
	}
	private String beforeTime;
	public String beforeTime() {
		return this.beforeTime;
	}
	private Long currencyGreaterThan;
	public Long currencyGreaterThan() {
		return this.currencyGreaterThan;
	}
	private Long currencyLessThan;
	public Long currencyLessThan() {
		return this.currencyLessThan;
	}
	private Boolean excludeCloseTo;
	public Boolean excludeCloseTo() {
		return this.excludeCloseTo;
	}
	private Long limit;
	public Long limit() {
		return this.limit;
	}
	private Long maxRound;
	public Long maxRound() {
		return this.maxRound;
	}
	private Long minRound;
	public Long minRound() {
		return this.minRound;
	}
	private String next;
	public String next() {
		return this.next;
	}
	private String notePrefix;
	public String notePrefix() {
		return this.notePrefix;
	}
	private Long round;
	public Long round() {
		return this.round;
	}
	private String sigType;
	public String sigType() {
		return this.sigType;
	}
	private String txId;
	public String txId() {
		return this.txId;
	}
	private String txType;
	public String txType() {
		return this.txType;
	}

	public SearchForTransactions(Client client) {
		super(client, "get");
	}

	/**
	 * Only include transactions with this address in one of the transaction fields. 
	 */
	public SearchForTransactions address(String address) {
		this.address = address;
		addQuery("address", String.valueOf(address));
		return this;
	}

	/**
	 * Combine with the address parameter to define what type of address to search for. 
	 */
	public SearchForTransactions addressRole(String addressRole) {
		this.addressRole = addressRole;
		addQuery("address-role", String.valueOf(addressRole));
		return this;
	}

	/**
	 * Include results after the given time. Must be an RFC 3339 formatted string. 
	 */
	public SearchForTransactions afterTime(String afterTime) {
		this.afterTime = afterTime;
		addQuery("after-time", String.valueOf(afterTime));
		return this;
	}

	/**
	 * Asset ID 
	 */
	public SearchForTransactions assetId(Long assetId) {
		this.assetId = assetId;
		addQuery("asset-id", String.valueOf(assetId));
		return this;
	}

	/**
	 * Include results before the given time. Must be an RFC 3339 formatted string. 
	 */
	public SearchForTransactions beforeTime(String beforeTime) {
		this.beforeTime = beforeTime;
		addQuery("before-time", String.valueOf(beforeTime));
		return this;
	}

	/**
	 * Results should have an amount greater than this value. MicroAlgos are the 
	 * default currency unless an asset-id is provided, in which case the asset will be 
	 * used. 
	 */
	public SearchForTransactions currencyGreaterThan(Long currencyGreaterThan) {
		this.currencyGreaterThan = currencyGreaterThan;
		addQuery("currency-greater-than", String.valueOf(currencyGreaterThan));
		return this;
	}

	/**
	 * Results should have an amount less than this value. MicroAlgos are the default 
	 * currency unless an asset-id is provided, in which case the asset will be used. 
	 */
	public SearchForTransactions currencyLessThan(Long currencyLessThan) {
		this.currencyLessThan = currencyLessThan;
		addQuery("currency-less-than", String.valueOf(currencyLessThan));
		return this;
	}

	/**
	 * Combine with address and address-role parameters to define what type of address 
	 * to search for. The close to fields are normally treated as a receiver, if you 
	 * would like to exclude them set this parameter to true. 
	 */
	public SearchForTransactions excludeCloseTo(Boolean excludeCloseTo) {
		this.excludeCloseTo = excludeCloseTo;
		addQuery("exclude-close-to", String.valueOf(excludeCloseTo));
		return this;
	}

	/**
	 * Maximum number of results to return. 
	 */
	public SearchForTransactions limit(Long limit) {
		this.limit = limit;
		addQuery("limit", String.valueOf(limit));
		return this;
	}

	/**
	 * Include results at or before the specified max-round. 
	 */
	public SearchForTransactions maxRound(Long maxRound) {
		this.maxRound = maxRound;
		addQuery("max-round", String.valueOf(maxRound));
		return this;
	}

	/**
	 * Include results at or after the specified min-round. 
	 */
	public SearchForTransactions minRound(Long minRound) {
		this.minRound = minRound;
		addQuery("min-round", String.valueOf(minRound));
		return this;
	}

	/**
	 * The next page of results. Use the next token provided by the previous results. 
	 */
	public SearchForTransactions next(String next) {
		this.next = next;
		addQuery("next", String.valueOf(next));
		return this;
	}

	/**
	 * Specifies a prefix which must be contained in the note field. 
	 */
	public SearchForTransactions notePrefix(String notePrefix) {
		this.notePrefix = notePrefix;
		addQuery("note-prefix", String.valueOf(notePrefix));
		return this;
	}

	/**
	 * Include results for the specified round. 
	 */
	public SearchForTransactions round(Long round) {
		this.round = round;
		addQuery("round", String.valueOf(round));
		return this;
	}

	/**
	 * SigType filters just results using the specified type of signature: * sig - 
	 * Standard * msig - MultiSig * lsig - LogicSig 
	 */
	public SearchForTransactions sigType(String sigType) {
		this.sigType = sigType;
		addQuery("sig-type", String.valueOf(sigType));
		return this;
	}

	/**
	 * Lookup the specific transaction by ID. 
	 */
	public SearchForTransactions txId(String txId) {
		this.txId = txId;
		addQuery("tx-id", String.valueOf(txId));
		return this;
	}
	public SearchForTransactions txType(String txType) {
		this.txType = txType;
		addQuery("tx-type", String.valueOf(txType));
		return this;
	}

	@Override
	public Response<TransactionsResponse> execute() throws Exception {
		Response<TransactionsResponse> resp = baseExecute();
		resp.setValueType(TransactionsResponse.class);
		return resp;
	}
	protected QueryData getRequestString() {
		addPathSegment(String.valueOf("transactions"));

		return qd;
	}
}

File: generator/go-sdk/indexer/searchForTransactions.go
package indexer

import (
	"context"
	"encoding/base64"
	"time"

	"github.com/algorand/go-algorand-sdk/client/v2/common"
	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
	"github.com/algorand/go-algorand-sdk/types"
)

// SearchForTransactions /v2/transactions
// Search for transactions.
type SearchForTransactions struct {
	c *Client
	p models.SearchForTransactionsParams
}

// Address only include transactions with this address in one of the transaction
// fields.
func (s *SearchForTransactions) Address(address types.Address) *SearchForTransactions {
	s.p.Address = address.String()
	return s
}

// AddressRole combine with the address parameter to define what type of address to
// search for.
func (s *SearchForTransactions) AddressRole(addressRole string) *SearchForTransactions {
	s.p.AddressRole = addressRole
	return s
}

// AfterTime include results after the given time. Must be an RFC 3339 formatted
// string.
func (s *SearchForTransactions) AfterTime(afterTime time.Time) *SearchForTransactions {
	s.p.AfterTime = afterTime.Format(time.RFC3339)
	return s
}

// ApplicationId application ID
func (s *SearchForTransactions) ApplicationId(applicationId uint64) *SearchForTransactions {
	s.p.ApplicationId = applicationId
	return s
}

// AssetID asset ID
func (s *SearchForTransactions) AssetID(assetID uint64) *SearchForTransactions {
	s.p.AssetID = assetID
	return s
}

// BeforeTime include results before the given time. Must be an RFC 3339 formatted
// string.
func (s *SearchForTransactions) BeforeTime(beforeTime time.Time) *SearchForTransactions {
	s.p.BeforeTime = beforeTime.Format(time.RFC3339)
	return s
}

// CurrencyGreaterThan results should have an amount greater than this value.
// MicroAlgos are the default currency unless an asset-id is provided, in which
// case the asset will be used.
func (s *SearchForTransactions) CurrencyGreaterThan(currencyGreaterThan uint64) *SearchForTransactions {
	s.p.CurrencyGreaterThan = currencyGreaterThan
	return s
}

// CurrencyLessThan results should have an amount less than this value. MicroAlgos
// are the default currency unless an asset-id is provided, in which case the asset
// will be used.
func (s *SearchForTransactions) CurrencyLessThan(currencyLessThan uint64) *SearchForTransactions {
	s.p.CurrencyLessThan = currencyLessThan
	return s
}

// ExcludeCloseTo combine with address and address-role parameters to define what
// type of address to search for. The close to fields are normally treated as a
// receiver, if you would like to exclude them set this parameter to true.
func (s *SearchForTransactions) ExcludeCloseTo(excludeCloseTo bool) *SearchForTransactions {
	s.p.ExcludeCloseTo = excludeCloseTo
	return s
}

// Limit maximum number of results to return.
func (s *SearchForTransactions) Limit(limit uint64) *SearchForTransactions {
	s.p.Limit = limit
	return s
}

// MaxRound include results at or before the specified max-round.
func (s *SearchForTransactions) MaxRound(maxRound uint64) *SearchForTransactions {
	s.p.MaxRound = maxRound
	return s
}

// MinRound include results at or after the specified min-round.
func (s *SearchForTransactions) MinRound(minRound uint64) *SearchForTransactions {
	s.p.MinRound = minRound
	return s
}

// Next the next page of results. Use the next token provided by the previous
// results.
func (s *SearchForTransactions) Next(next string) *SearchForTransactions {
	s.p.Next = next
	return s
}

// NotePrefix specifies a prefix which must be contained in the note field.
func (s *SearchForTransactions) NotePrefix(notePrefix []byte) *SearchForTransactions {
	s.p.NotePrefix = base64.StdEncoding.EncodeToString(notePrefix)
	return s
}

// RekeyTo include results which include the rekey-to field.
func (s *SearchForTransactions) RekeyTo(rekeyTo bool) *SearchForTransactions {
	s.p.RekeyTo = rekeyTo
	return s
}

// Round include results for the specified round.
func (s *SearchForTransactions) Round(round uint64) *SearchForTransactions {
	s.p.Round = round
	return s
}

// SigType filters just results using the specified type of signature:
// * sig - Standard
// * msig - MultiSig
// * lsig - LogicSig
func (s *SearchForTransactions) SigType(sigType string) *SearchForTransactions {
	s.p.SigType = sigType
	return s
}

func (s *SearchForTransactions) TxType(txType string) *SearchForTransactions {
	s.p.TxType = txType
	return s
}

// TxID lookup the specific transaction by ID.
func (s *SearchForTransactions) TxID(txID string) *SearchForTransactions {
	s.p.TxID = txID
	return s
}

// Do performs HTTP request
func (s *SearchForTransactions) Do(ctx context.Context,
	headers ...*common.Header) (response models.TransactionsResponse, err error) {
	err = s.c.get(ctx, &response,
		"/v2/transactions", s.p, headers)
	return
}
